pkg/services/ngalert/notifier/channels/sensugo.go

```diff
@@ -0,0 +1,201 @@
+package channels
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"path"
+	"strings"
+	"time"
+
+	gokit_log "github.com/go-kit/kit/log"
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/alerting"
+	old_notifiers "github.com/grafana/grafana/pkg/services/alerting/notifiers"
+	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/model"
+)
+
+type SensuGoNotifier struct {
+	old_notifiers.NotifierBase
+	log  log.Logger
+	tmpl *template.Template
+
+	URL       string
+	Entity    string
+	Check     string
+	Namespace string
+	Handler   string
+	APIKey    string
+	Message   string
+}
+
+// NewSlackNotifier is the constructor for the Slack notifier
+func NewSensuGoNotifier(model *models.AlertNotification, t *template.Template) (*SensuGoNotifier, error) {
+	if model.Settings == nil {
+		return nil, alerting.ValidationError{Reason: "No Settings Supplied"}
+	}
+
+	url := model.Settings.Get("url").MustString()
+	if url == "" {
+		return nil, alerting.ValidationError{Reason: "Could not find URL property in settings"}
+	}
+
+	apikey := model.DecryptedValue("apikey", model.Settings.Get("apikey").MustString())
+	if apikey == "" {
+		return nil, alerting.ValidationError{Reason: "Could not find the API Key property in settings"}
```

Comment:
There's no reason to capitalize key, right (the JSON property is "apikey" anyway):
```suggestion
		return nil, alerting.ValidationError{Reason: "Could not find the API key property in settings"}
```